### PR TITLE
feat(CMakeLists.txt): Code coverage added on unit tests.

### PR DESCRIPTION
**Pull request by bouda1**
The cmake option WITH_COVERAGE=On must be provided in complement to
WITH_TESTING=On.

Then, the command _make test-coverage_ executes the tests and fills
a directory 'coverage' containing the coverage.

### CODE REVIEW

**Comment by bouda1** [comment]
Updated.

File: build/CMakeLists.txt [hunk 1]
```diff
@@ -1744,6 +1752,9 @@ if (WITH_TESTING)
   # target_link_libraries("${TEST_NAME}" "roker")
   # add_test("${TEST_NAME}" "${TEST_NAME}")
 
+  enable_testing()
```
Comment (ganoze) [inline build/CMakeLists.txt]: Please move around line 1284, juste after if (WITH_TESTING).

_if (WITH_TESTING)
  enable_testing()
  add_subdirectory("tests")_

File: build/CMakeLists.txt [hunk 2]
```diff
@@ -1744,6 +1752,9 @@ if (WITH_TESTING)
...
+  add_test(tests tests/ut)
```
Comment (ganoze) [inline build/CMakeLists.txt]: Please move to build/tests/CMakeLists.txt.

File: build/tests/CMakeLists.txt [hunk 3]
```diff
@@ -48,3 +48,14 @@ add_executable("ut"
   ${TESTS_SOURCES}
 )
 target_link_libraries("ut" "gtest" ${TESTS_LIBRARIES} ${NEB} "rokerbase" "roker")
+
+add_test(tests ut)
```
Comment (ganoze) [inline build/tests/CMakeLists.txt]: Please use add_test(NAME tests COMMAND ut) for the automatic binary path replacement (https://cmake.org/cmake/help/v3.9/command/add_test.html).

**Comment by bouda1** [comment]
Should be OK now.

**Pull request merged**